Content.Server/GameTicking/GamePreset.cs

```diff
@@ -18,6 +31,80 @@ public abstract class GamePreset
 
         public virtual void OnGameStarted() { }
 
+        /// <summary>
+        /// Called when a player is spawned in (this includes, but is not limited to, before Start)
+        /// </summary>
+        public virtual void OnSpawnPlayerCompleted(IPlayerSession session, IEntity mob, bool lateJoin) { }
+
+        /// <summary>
+        /// Called when a player attempts to ghost.
+        /// </summary>
+        public virtual void OnGhostAttempt(IConsoleShell? shell, IPlayerSession player, bool canReturnGlobal)
+        {
+            var mind = player.ContentData().Mind;
+            if (mind == null)
+            {
+                shell?.SendText(player, "You can't ghost here!");
+                return;
+            }
```

Comment:
i agree with sloth, this part should be handled outside, the function should only receive a `Mind` (and the `canReturnGlobal` flag of course)

Reply:
Done.